app/src/main/java/org/novinger/jason/demoreader/ListActivity.java

```diff
@@ -67,4 +90,22 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
         return super.onOptionsItemSelected(item);
     }
+
+    private Article[] getFeedItems(String url) throws IOException {
```

Comment:
again, ref the note above about primitive arrays - they might be appropriate here, and if it's a deliberate choice, +100, but understand the limits vs `Collections` (which are `Lists` and `Sets`).

Personally, I use `HashSet` the most, then `LinkedList`, then `ArrayList`.  `HashSet` is super efficient, O(1) on most ops, but obv it's a `Set` so no duplicates